src/triple.rs

```diff
@@ -127,6 +127,38 @@ impl Triple {
             _ => return Err(()),
         })
     }
+
+    /// The C data model for a given target. If the model is not known, returns `Err(())`.
+    pub fn data_model(&self) -> Result<CDataModel, ()> {
+        match self.pointer_width()? {
+            PointerWidth::U64 => {
+                if self.operating_system == OperatingSystem::Windows {
+                    Ok(CDataModel::LLP64)
+                } else if self.default_calling_convention() == Ok(CallingConvention::SystemV)
+                    || self.architecture == Architecture::Wasm64 {
+                    Ok(CDataModel::LP64)
+                } else {
+                    Err(())
+                }
+            }
+            PointerWidth::U32 => {
+                if self.operating_system == OperatingSystem::Windows
+                    || self.default_calling_convention() == Ok(CallingConvention::SystemV)
+                    || self.architecture == Architecture::Wasm32 {
+                    Ok(CDataModel::ILP32)
+                } else {
+                    Err(())
+                }
+            }
+            PointerWidth::U16 => {
+                if self.operating_system == OperatingSystem::Windows {
+                    Ok(CDataModel::LP32)
```

Comment:
According to https://en.m.wikipedia.org/wiki/16-bit_computing there is a difference between near (16bit) and far (32bit) pointers on 16bit x86. There are multiple code models that use a different mix of near and far pointers for code and data.

Reply:
This sounds like a little more than I want to tackle in a first PR, especially since target_lexicon doesn't currently distinguish between code and data pointers. Are you good with me removing this U16 branch altogether and just returning `Err`?

Reply:
Yes, that sounds good.